### PR TITLE
gh-116869: Add test_cext test: build a C extension

### PR DESCRIPTION
**Pull request by vstinner**
* Issue: gh-116869

### CODE REVIEW

**Comment by vstinner** [comment]
Once my fix PR gh-116950 will be merged, I will write a second change on top of this issue to run tests with `-Werror=declaration-after-statement` to avoid regressions.

**Comment by vstinner** [comment]
Oh, the test catched a warning when using `-std=c99`:

```
      In file included from extension.c:7:
      In file included from /buildbot/buildarea/pull_request.ware-freebsd/build/Include/Python.h:122:
      /buildbot/buildarea/pull_request.ware-freebsd/build/Include/cpython/optimizer.h:77:3: error: redefinition of typedef '_PyOptimizerObject' is a C11 feature [-Werror,-Wtypedef-redefinition]
      } _PyOptimizerObject;
        ^
      /buildbot/buildarea/pull_request.ware-freebsd/build/Include/cpython/optimizer.h:60:35: note: previous definition is here
      typedef struct _PyOptimizerObject _PyOptimizerObject;
                                        ^
      1 error generated.
```

I wrote PR https://github.com/python/cpython/pull/116963 to fix it.

**Comment by vstinner** [comment]
PR rebased on top of PR #116963 merged.

File: Lib/test/test_cext/__init__.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+
+SOURCE = os.path.join(os.path.dirname(__file__), 'extension.c')
+SETUP = os.path.join(os.path.dirname(__file__), 'setup.py')
+
+
+# gh-110119: pip does not currently support 't' in the ABI flag use by
+# --disable-gil builds. Once it does, we can remove this skip.
+@unittest.skipIf(support.Py_GIL_DISABLED,
+                 'test does not work with --disable-gil')
```
Comment (colesbury) [inline Lib/test/test_cext/__init__.py:19]: It should work now that https://github.com/python/cpython/pull/116878 landed!

(The comment was a bit inaccurate: it turns out the issue was in `wheel`, not `pip`)

Comment (vstinner) [reply Lib/test/test_cext/__init__.py:19]: That's great! I will write a follow-up to patch also test_cppext

**Comment by vstinner** [comment]
Let's start with something simple.

File: Lib/test/test_cext/__init__.py (hunk 1 above)
Comment (vstinner) [reply Lib/test/test_cext/__init__.py:19]: PR https://github.com/python/cpython/pull/116973

**Pull request merged**